bin/get-video-s3-acls.py

```diff
@@ -0,0 +1,418 @@
+
+
+# Pull all info from existing cache file
+def get_keys_from_cache_file():
+    this_all_keys_dict = {}
+    with open(ALL_KEYS_CACHE_FILE, "r") as cache_file:
+        for line in cache_file.readlines():
+            (
+                video_key,
+                key_in_nzsl_str,
+                key_in_s3_str,
+                gloss_id_str,
+                gloss_idgloss,
+                created_at,
+                gloss_public_str,
+                video_public_str,
+                video_id_str,
+            ) = line.strip().split(CSV_DELIMITER)
+
+            key_in_nzsl = key_in_nzsl_str.strip().lower() == "true"
+            key_in_s3 = key_in_s3_str.strip().lower() == "true"
+            if key_in_nzsl:
+                video_id = int(video_id_str)
+                # Some have no gloss_id
+                try:
+                    gloss_id = int(gloss_id_str)
+                except ValueError:
+                    gloss_id = None
+                gloss_public = gloss_public_str.strip().lower() == "true"
+                video_public = video_public_str.strip().lower() == "true"
+            else:
+                video_id = ""
+                gloss_id = ""
+                gloss_public = ""
+                video_public = ""
+
+            this_all_keys_dict[video_key] = [
+                key_in_nzsl,
+                key_in_s3,
+                video_id,
+                gloss_id,
+                gloss_public,
+                video_public,
+            ]
+
+        return this_all_keys_dict
+
+
+# Get all keys from AWS S3
+def get_s3_bucket_raw_keys_list(s3_bucket=AWS_S3_BUCKET):
+    print(f"Getting raw AWS S3 keys recursively ({s3_bucket}) ...", file=sys.stderr)
+    result = subprocess.run(
+        [
+            AWSCLI,
+            "s3",
+            "ls",
+            f"s3://{s3_bucket}",
+            "--recursive",
+        ],
+        env=os.environ,
+        capture_output=True,
+        check=True,
+        text=True,
+    )
+
+    # Separate out just the key from date, time, size, key
+    this_s3_bucket_raw_keys_list = []
+    for line in result.stdout.split("\n"):
+        if line:
+            this_s3_bucket_raw_keys_list.append(re.split(r"\s+", line, 3)[3])
+
+    print(
+        f"{len(this_s3_bucket_raw_keys_list)} rows retrieved",
+        file=sys.stderr,
+    )
+
+    return this_s3_bucket_raw_keys_list
+
+
+# Get the video files info from NZSL Signbank
+def get_nzsl_raw_keys_dict():
+    print(
+        f"Getting raw list of video file info from NZSL Signbank ...",
+        file=sys.stderr,
+    )
+    this_nzsl_raw_keys_dict = {}
+    # Column renaming is purely for readability
+    # Special delimiter because columns might contain commas
+    result = subprocess.run(
+        [
+            PGCLI,
+            "-c",
+            "COPY ("
+            "SELECT "
+            "dg.id AS gloss_id, "
+            "dg.idgloss AS gloss_idgloss, "
+            "dg.created_at, "
```

Comment:
issue: while I'm not against including this `created_at`, the more important one that we actually wanted is for the S3 file

Reply:
It doesn't appear to have one 8|
I'll look deeper

Reply:
we can get `Last-Modified` from the metadata, and there should be a "created at" (or "uploaded at" or something) stamp somewhere around - start with `Last-Modified` as that should be easy

Reply:
Ah, found the relevant tidbit:
https://docs.aws.amazon.com/AmazonS3/latest/userguide/intro-lifecycle-rules.html

So, yes, I'll use the LastModified field.

```
Note
Amazon S3 maintains only the last modified date for each object. For example, the Amazon S3 console shows the Last modified date in the object's Properties pane. When you initially create a new object, this date reflects the date that the object is created. If you replace the object, the date changes accordingly. Therefore, the creation date is synonymous with the Last modified date.
```

Reply:
Done.